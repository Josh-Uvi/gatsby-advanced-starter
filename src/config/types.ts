export interface WebsiteData {
  title: string;
  titleShort: string;
  name: string;
  description: string;
  logoUrl: string;
  fbAppId?: string;
  twitterName?: string;
  url: string;
  rss: string;
  rssTitle: string;
}

export interface UserData {
  id: string;
  firstName: string;
  lastName: string;
  twitterName?: string;
  email: string;
  location: string;
  avatar: string;
  links?: {
    label: string;
    url: string;
  }[];
}

export interface OrganizationData {
  name: string;
  description: string;
  logoUrl: string;
  url: string;
}

export interface SiteConfig {
  website: WebsiteData;
  user: UserData;
  organization: OrganizationData;
  pathPrefix: string;
  googleAnalyticsID?: string;
  disqusShortname?: string;
  postsPerPage: number;
  copyright: string;
  themeColor: string;
  backgroundColor: string;
}
