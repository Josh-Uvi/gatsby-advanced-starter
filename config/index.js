const validateSiteConfig = require("./Validator");
const userConfig = require("../data/SiteConfig.js");

module.exports = validateSiteConfig(userConfig);
