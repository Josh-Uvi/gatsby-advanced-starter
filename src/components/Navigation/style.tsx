import styled from "styled-components";

import { Link } from "gatsby";
import { styles } from "../../theme";

export const HomeButton = styled(Link)`
  text-decoration: none;
  color: var(--color-text);

  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
  align-items: center;
`;

export const NavigationContainer = styled.header`
  display: flex;
  flex-direction: row;

  justify-content: space-between;
  align-items: center;
`;

export const ButtonContainer = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-gap: 24px;

  align-items: flex-start;
`;

export const SiteTitle = styled.p`
  ${styles.ButtonLabel}

  @media (max-width: 500px) {
    display: none;
  }
`;

type StyledNavButtonProps = {
  activeClassName: string;
};

export const StyledNavButton = styled(Link).attrs(
  ({ activeClassName }: StyledNavButtonProps) => ({
    activeClassName,
  })
)`
  ${styles.ButtonLabel}

  padding: 0 0 8px 0;

  background-color: var(--color-background);

  border-style: solid;
  border-color: var(--color-primary);

  border-width: 0px;

  &.${(props) => props.activeClassName} {
    border-bottom-width: 2px;
  }

  text-decoration: none;
  color: var(--color-text);
`;
