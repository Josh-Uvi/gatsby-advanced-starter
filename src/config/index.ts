import validateSiteConfig from "./Validator";
import baseConfig from "./defaultConfig";

const validatedConfig = validateSiteConfig(baseConfig);

export type {
  SiteConfig,
  OrganizationData,
  UserData,
  WebsiteData,
} from "./types";

export { default as schema } from "./schema";
export { default as constants } from "./constants";

export const defaultConfig = validatedConfig;
