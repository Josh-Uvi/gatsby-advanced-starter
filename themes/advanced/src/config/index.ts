import urlJoin from "url-join";
import { merge } from "lodash";

import validateSiteConfig from "./Validator";
import baseConfig from "./defaultConfig";
import { SiteConfig } from "./types";

const validatedConfig = validateSiteConfig(baseConfig);

export type {
  SiteConfig,
  OrganizationData,
  UserData,
  WebsiteData,
} from "./types";

export { default as schema } from "./schema";
export { default as constants } from "./constants";

export const defaultConfig = validatedConfig;

export const withDefaults = (userConfig: SiteConfig): Readonly<SiteConfig> =>
  merge(defaultConfig, userConfig);

export const withBasePath = (config: SiteConfig, url: string): string =>
  config.basePath ? urlJoin(config.basePath, url) : url;
