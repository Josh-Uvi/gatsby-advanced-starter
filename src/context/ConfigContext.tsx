import * as React from "react";
import config from "../config";

const ConfigContext = React.createContext(config);

const ConfigProvider: React.FC = ({ children }) => (
  <ConfigContext.Provider value={config}>{children}</ConfigContext.Provider>
);

export default ConfigContext;
export { ConfigProvider };
