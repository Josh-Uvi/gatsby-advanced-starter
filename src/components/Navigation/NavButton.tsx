import * as React from "react";

import { StyledNavButton } from "./style";

type NavButtonProps = {
  href: string;
  children?: JSX.Element[] | string;
};

const NavButton = ({ children, href }: NavButtonProps): JSX.Element => (
  <StyledNavButton activeClassName="gatsby-link-active" to={href}>
    {children}
  </StyledNavButton>
);

export default NavButton;
