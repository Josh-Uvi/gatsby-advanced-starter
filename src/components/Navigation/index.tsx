import React, { useContext } from "react";

import NavButton from "./NavButton";
import { AdvancedLogo } from "../../icons";
import ConfigContext from "../../context/ConfigContext";

import * as S from "./style";

const Navigation = (): JSX.Element => {
  const config = useContext(ConfigContext);

  return (
    <S.Wrapper>
      <S.HomeButton to="/">
        <AdvancedLogo width={36} height={36} />
        <S.SiteTitle>{config.website.titleShort}</S.SiteTitle>
      </S.HomeButton>
      <S.NavGrid>
        <NavButton href="/">Posts</NavButton>
        <NavButton href="/about">About</NavButton>
      </S.NavGrid>
    </S.Wrapper>
  );
};

export default Navigation;
