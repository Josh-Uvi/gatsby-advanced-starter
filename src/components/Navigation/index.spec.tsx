import React from "react";
import "jest-styled-components";
import { render } from "@testing-library/react";

import Navigation from ".";

import ConfigContext from "../../context/ConfigContext";
import config from "../../config";

jest.mock("../../config");

describe("component Navigation", () => {
  it("renders correctly", () => {
    expect.assertions(1);

    const { asFragment } = render(
      <ConfigContext.Provider value={config}>
        <Navigation />
      </ConfigContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
