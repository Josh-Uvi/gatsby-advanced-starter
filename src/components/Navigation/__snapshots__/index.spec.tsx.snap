// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Navigation renders correctly 1`] = `
<DocumentFragment>
  .c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--color-text);
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: grid;
  grid-auto-flow: column;
  grid-gap: 24px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 16px;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  text-transform: uppercase;
}

.c4 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 16px;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  text-transform: uppercase;
  padding: 0 0 8px 0;
  background-color: var(--color-background);
  border-style: solid;
  border-color: var(--color-primary);
  border-width: 0px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--color-text);
}

.c4.gatsby-link-active {
  border-bottom-width: 2px;
}

@media (max-width:500px) {
  .c2 {
    display: none;
  }
}

<header
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      <test-file-stub
        height="36"
        width="36"
      />
      <p
        class="c2"
      >
        Advanced Blog
      </p>
    </a>
    <div
      class="c3"
    >
      <a
        class="c4"
        href="/"
      >
        Posts
      </a>
      <a
        class="c4"
        href="/about"
      >
        About
      </a>
    </div>
  </header>
</DocumentFragment>
`;
